### PR TITLE
[이원규-10주차 알고리즘 스터디]

### PR DESCRIPTION
**Pull request by OneK-2**
### Week 10 (24.09.26 ~ 24.10.02)
| Category | Title | Link |
| :------: | :---: | :--: |
| 구현 |  <a href="https://www.acmicpc.net/problem/15787">기차가 어둠을 헤치고 은하수를</a> | <a href="https://github.com/Algorithm-Study-AS/Algorithm-study/commit/ce7709d27a02f7e4b69f75e4e3f5f53c245f8b4a">🔗</a> |
| 구현 |  <a href="https://www.acmicpc.net/problem/1913">달팽이</a> | <a href="https://github.com/Algorithm-Study-AS/Algorithm-study/commit/1a1516c64dda3bf32cfcf8a56d803e50ce08d9b6">🔗</a> |
| DP |  <a href="https://www.acmicpc.net/problem/2579">계단 오르기</a> | <a href="https://github.com/Algorithm-Study-AS/Algorithm-study/commit/bf0915aca392e6f24df3d5ccd35ab8b1b81fdf8a">🔗</a> |
| DP |  <a href="https://www.acmicpc.net/problem/11053">가장 긴 증가하는 부분 수열</a> | <a href="https://github.com/Algorithm-Study-AS/Algorithm-study/commit/a68fd5bcce6399b37e803e8e99fea48325e7fa54">🔗</a> |
<br>

```
To. Reviewers
```

### CODE REVIEW

File: 이원규/DP/계단 오르기-2579.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            arr[i] = Integer.parseInt(br.readLine());
+        }
+        dp[0] = 0;
+        dp[1] = arr[1];
+        if (n >= 2) {
+            dp[2] = arr[2] + arr[1];
+        }
+
+        for (int i = 3; i <= n; i++) {
+            dp[i] = Math.max(dp[i - 2], dp[i - 3] + arr[i - 1]) + arr[i];
```
Comment (harin1212) [inline 이원규/DP/계단 오르기-2579.java:24]: 저랑 동일합니다!

File: 이원규/구현/기차가 어둠을 헤치고 은하수를-15787.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                }
+                train[trainNum][20] = 0;
+            }
+        }
+        // 중복체크
+        Set<String> set = new HashSet<>();
+        for (int i = 1; i <= n; i++) {
+            StringBuilder sb = new StringBuilder();
+            for (int j = 1; j <= 20; j++) {
+                sb.append(train[i][j]);
```
Comment (harin1212) [inline 이원규/구현/기차가 어둠을 헤치고 은하수를-15787.java:44]: 같은 배열 판별하는 다른 방법 아실까요..? 저도 이렇게 풀었습니다

File: 이원규/구현/기차가 어둠을 헤치고 은하수를-15787.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * N(1 ≤ N ≤ 100000)과 명령의 수 M(1 ≤ M ≤ 100000)
+ * 1 i x : i번째 기차에(1 ≤ i ≤ N) x번째 좌석에(1 ≤ x ≤ 20) 사람 탑승
+ * 타있으면 아무것도 안함
+ * 2 i x : i번째 기차에 x번째 좌석에 앉은 사람은 하차
+ * 3 i : i번째 기차에 앉아있는 승객들이 모두 한칸씩 뒤로
+ * 만약 20번째 자리에 사람이 앉아있었다면 그 사람은 이 명령 후에 하차
+ * 4 i : i번째 기차에 앉아있는 승객들이 모두 한칸씩 앞으로
+ * 만약 1번째 자리에 사람이 앉아있었다면 그 사람은 이 명령 후에 하차
+ * <p>
+ * 중복체크 어떻게 할지 고민함.. set에 배열 넣어봤는데 실패
```
Comment (harin1212) [inline 이원규/구현/기차가 어둠을 헤치고 은하수를-15787.java:65]: ㅋㅋㅋ 저도... 똑같은 흐름

File: 이원규/DP/가장 긴 증가하는 부분 수열-11053.java
Comment (harin1212) [inline 이원규/DP/가장 긴 증가하는 부분 수열-11053.java:1]: 저랑 비슷합니다~

**Review by harin1212 — APPROVED** [review]

File: 이원규/구현/기차가 어둠을 헤치고 은하수를-15787.java (hunk 2 above)
Comment (OneK-2) [reply 이원규/구현/기차가 어둠을 헤치고 은하수를-15787.java:44]: 이게 최선이었던것같습니다..

**Pull request merged**